### PR TITLE
Fix the issue where AbsFOp gets i16 type.

### PR DESCRIPTION
**Pull request by Stonepia**
When the frontend type is `bf16`, the triton will store it as `i16`. This makes AbsFOp complain about

```
error: 'spirv.CL.fabs' op operand #0 must be 16/32/64-bit float or vector of 16/32/64-bit float values of length 2/3/4/8/16, but got 'i16'
```

This PR adds the conversion to support the above case.

### CODE REVIEW

File: lib/Conversion/TritonGPUToSPIRV/ElementwiseOpToSPIRV.cpp [hunk 1]
```diff
@@ -948,6 +948,33 @@ struct ExpOpSPIRVConversionApprox
...
+  using Base =
+      ElementwiseOpSPIRVConversionBase<mlir::math::AbsFOp, AbsFOpConversion>;
+  using Base::Base;
+  using Adaptor = typename Base::OpAdaptor;
+
+  Value createDestOp(mlir::math::AbsFOp op, OpAdaptor adaptor,
+                     ConversionPatternRewriter &rewriter, Type elemTy,
+                     ValueRange operands, Location loc) const {
+
+    if (llvm::isa<IntegerType>(elemTy)) {
```
Comment (chengjunlu) [inline lib/Conversion/TritonGPUToSPIRV/ElementwiseOpToSPIRV.cpp:962]: Check how cuda support this.

Comment (chengjunlu) [reply lib/Conversion/TritonGPUToSPIRV/ElementwiseOpToSPIRV.cpp:962]: It should be common in ElementwiseOpSPIRVConversion. Right?

Comment (Stonepia) [reply lib/Conversion/TritonGPUToSPIRV/ElementwiseOpToSPIRV.cpp:962]: Thanks for the comment! This aligns with CUDA implementation:
https://github.com/openai/triton/blob/main/lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp#L1015-L1030

Comment (chengjunlu) [reply lib/Conversion/TritonGPUToSPIRV/ElementwiseOpToSPIRV.cpp:962]: Fair enough. Let's use this for now.
refactor in future if it becomes common issue.

**Pull request merged**